src/common.d.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, PureComponent, FC } from "react"
+
+export declare type SubscribeFn<T> = (data: T) => void
+
+export declare type UnsubscribeFn = () => void
+
+export declare type MapFn<T> = (data: T) => T | any
+
+export declare type GenericReactComponent<T, S, SS> = (
+  Component<T, S, SS> | PureComponent<T, S, SS> | FC<T>
+)
+
+export declare type ComponentWrapper<T, S, SS> = (
+  (component: GenericReactComponent<T, S, SS>) => Component<T>
+)
+
+export declare interface Store<T> {
+  getState(): T
+  set(data: Partial<T>): void
+  observer<S, SS>(options: ObserverOptions<T>): Observer<T, S, SS>
```

Comment:
Since there are more than 2 types in generics I would suggest using full names instead of `T`, `S` and `SS`. 
Otherwise, it's quite difficult to read and understand what is what :)

Reply:
True. Though I didn't wrote full type names for all `multipleObserver` function overloads, not going to help much as I think

By the way, is it ok that  it's only 6 overloads? People probably don't tend to use more than that but still, I don't really like overloading technique here and I wonder if there's any procedural way to infer types from generic array of args...

Reply:
Yeah, I think that's enough 👍